[Justin Gnoh] iP

justgnohUG opened the pull request.

Commit:
Add Gradle support

Commit:
build.gradle: Update version to 8.29

justgnohUG committed:
Level 2: Add, List

justgnohUG committed:
Level 3: Mark as Done

justgnohUG committed:
Level 4 ToDo,Event,Deadline inheritance from Task

justgnohUG committed:
Testing using the I/O redirection technique

- Take note that .trim() has been added to each case
- homebrew install dos2unix
- Take note of trailing space from .split()

justgnohUG committed:
Level-5: Handle Errors

- Created DukeException.java
- Done command lacks error handling
- Handles Empty Descriptions
- Handles unknown commands
- Any unknown commands get sent to default

justgnohUG committed:
Level-5.1: Added Exception handling in Duke.java

- Changed to Switch statements
- Note that this portion was manually staged for this commit
- Had to remove Level-6 Contents for this commit

justgnohUG committed:
Level-6: Delete

- Added Delete command
- Updated runtest.sh input.txt

justgnohUG committed:
Level 1: Greet, Echo, Exit

Changed Level-1 author

Cheng20010201 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,13 @@
+public class Deadline extends Task {
```
LGTM. Perhaps add some javadoc for your public class/methods? (applicable to other classes as well) :D

Cheng20010201 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,154 @@
+        Scanner scan = new Scanner(System.in);
+        String input = " ";
+
+//        Initialize Task container
```
Maybe can allign the '//' to your code? Like:
```java
    // Initialize Task container
    ArrayList<Task> tasks = new ArrayList<>();
```

Cheng20010201 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,154 @@
+        int taskNumber = 0;
+
+
+        while(!input.equals("bye")) {
```
I would say adding an extra while space here?
```java
    while (!input.equals("bye")) { 
    ...
```

Cheng20010201 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,154 @@
+            input = scan.nextLine().trim();
+            String command = input.split(" ")[0];
+
+            switch (command) {
```
I guess can try to configure your IDEA to apply the same indentation for switch as in the coding standard?
```java
    switch (command) {
    case "bye":
        // ...
    case "list":
        // ...

```

Cheng20010201 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,154 @@
+                    System.out.println("\n---------------------------------------" );
+                    System.out.println("Here are the tasks in your list:");
+                    for (int i = 0; i < tasks.size(); i++) {
+                        System.out.println(String.valueOf(i+1) + "." + tasks.get(i).toString());
```
Wanna replace `i+1` with `i + 1`? trivial issue though. :)

Cheng20010201 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,154 @@
+                        System.out.println(todo.toString());
+                        System.out.println("Now you have " + tasks.size() + " tasks in the list");
+                        System.out.println("---------------------------------------" );
+                    } catch(DukeException e) {
```
I guess a white space can be placed here after `catch`, as per coding standards. 🤔

Cheng20010201 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,154 @@
+                    break;
+
+                case "deadline":
+                    try{
```
Same white space issue as above :)

Cheng20010201 reviewed (commented):
The code is succint in general and it does not have major violations with coding standard. There are, however, a few trivial white space and indentation issues that are easy to fix. Other than that, well done! :D

justgnohUG committed:
Level 7: Save

- Add Save functionality on "bye" command
- Creates new /data directory if folder doesn't exist

justgnohUG committed:
Level 7: Save

- Add Save functionality on "bye" command
- Creates new /data directory if folder doesn't exist

justgnohUG committed:
Level7.1: Save

- Add "load from file"
- Changed "done" booleans in Tasks and its children to "isDone"

justgnohUG committed:
Fix exception to return empty ArrayList on error

- fetchTask() exception handles by return empty ArrayList

justgnohUG committed:
Add date and time support for deadlines

- Add processDate(String date) static method

justgnohUG committed:
Modify Deadline /by property to LocalDate type

- Modify fetchTasks() to retrieve LocalDate
- Add date parsing in fetchTasks()

justgnohUG committed:
Merge pull request #1 from justgnohUG/branch-Level-7

Branch level 7

justgnohUG committed:
Merge branch 'branch-Level-7'

justgnohUG committed:
Merge branch 'master' of https://github.com/justgnohUG/ip

justgnohUG committed:
Merge branch 'master' into branch-Level-8

justgnohUG committed:
Merge pull request #2 from justgnohUG/branch-Level-8

Branch level 8

justgnohUG committed:
Fix Syntax error in Deadline.java

justgnohUG committed:
Use more OOP in Duke.java

- Add Parser.java to process commands
- Add Ui.java for user interaction
- Add TaskList.java to store ArrayList<Task> and task functionalities
- Add Storage.java to load() and save() tasks on local

justgnohUG committed:
Add JUnit Tests

- TaskListTest for addTask() and deleteTask()
- TodoTest for toString() conversion

justgnohUG committed:
Consolidate classes into "DukeTools" Package

- Deadline, Event, Todo, Task
- DukeException
- Parser, Ui, Storage, TaskList

justgnohUG committed:
Package the App as a JAR File

justgnohUG committed:
Add JavaDoc comments

justgnohUG committed:
Tweak existing user base to satisfy coding standard

- Fix comment indentations
- Set appropriate white space between characters
- Re-indented Switch and Case statements

justgnohUG committed:
Add "find" command

- Add getArguments(String input) to receive arguments after command
- Add printMatchingTask(String output)
- Add matchTasks(String arguments) to do task matching

justgnohUG committed:
Merge pull request #3 from justgnohUG/branch-A-JavaDoc

Add JavaDoc comments

justgnohUG committed:
Merge branch 'branch-A-CodingStandard'

justgnohUG committed:
Resolved merge conflict by fixing JavaDoc for Storage.java

justgnohUG committed:
Merge branch 'branch-Level-9'

justgnohUG committed:
Fix missing methods in Parser.java due to merging

- Fix due to merge "branch-Level-9"

justgnohUG committed:
Merge remote-tracking branch 'origin/add-gradle-support'

justgnohUG committed:
Configure mainClassName to run Duke.java

justgnohUG committed:
Modify code to adhere to checkstyle

justgnohUG committed:
Merge branch 'branch-A-CheckStyle'

justgnohUG committed:
Add JavaFX GUI

- Configure Duke logic to work with GUI
- Add try,catch for "delete" and "done" functions

justgnohUG committed:
Add TaskListTest.java exception handling

- This will not be required in the future after transitioning
to GUI

justgnohUG committed:
Add assertions in Duke

- Add assertions mainly in Duke.java
- Check for addTask() and deleteTask()
- Assert for getSingleTask()

justgnohUG committed:
Add CodeQuality improvements

justgnohUG committed:
Merge pull request #6 from justgnohUG/branch-A-Assertions

Branch-A-assertions

justgnohUG committed:
Merge branch 'master' into branch-A-CodeQuality

justgnohUG committed:
Merge pull request #7 from justgnohUG/branch-A-CodeQuality

Branch-a-Code Quality

justgnohUG committed:
Add JavaFX jar support

- Find() now returns task not found message
- Add Java Docs to new methods
- Add constants to remove "Magic Variables" Ui.java

justgnohUG committed:
Add introduction message on startup

- Add help message for "help" command

justgnohUG committed:
Add B-Snooze extension

- Deadline tasks can be snoozed to a new date
- Snooze command without "/to" attribute automatically postpones event
  by 1 day
- Updated Ui messages
- Add string-LocalDate conversion in Parser.java

justgnohUG committed:
Set theme jekyll-theme-minimal

justgnohUG committed:
Update DukeException messages

- DukeException printMessage()
- Add DukeException to getSingleTask()
- Add try-catch to methods affected by new DukeException

justgnohUG committed:
Merge branch 'branch-B-Extension'

justgnohUG committed:
Merge branch 'master' of github.com:justgnohUG/ip

justgnohUG committed:
Update Readme.md UserGuide

justgnohUG committed:
Update UserGuide

justgnohUG committed:
Update UserGuide

Add more changes to UserDoc

justgnohUG committed:
Merge remote-tracking branch 'refs/remotes/origin/master'

Conflicts:
	docs/README.md

justgnohUG committed:
Update markdown links in userguide

justgnohUG committed:
Improve Gui by adding CSS styling

- Add application.css
- Change user and duke image
- Change Color scheme

justgnohUG committed:
Add Ui.png

justgnohUG committed:
Merge branch 'branch-BetterGui'

justgnohUG committed:
Add more OOP and abstraction

- Add CommandHandler.java to handle commands
- Add Speech Bubble in GUI
- Add methods to get task attributes in Parser.java
- Add acknowledgements in README.md

justgnohUG committed:
Address Checkstyle issues

- Rectify Empty catch statement in matchTasks()

justgnohUG committed:
Add Java dependency in build.gradle
